support/controlplane-component/common.go

```diff
@@ -41,3 +41,13 @@ func EnableForPlatform(platform hyperv1.PlatformType) option {
 		return cpContext.HCP.Spec.Platform.Type == platform
 	})
 }
+
+// EnableIfCustomKubeconfig is a helper predicate for the common use case of enabling a resource when a KubeAPICustomKubeconfig is specified.
+func EnableIfCustomKubeconfig() option {
+	return WithPredicate(func(cpContext WorkloadContext) bool {
```

Comment:
I don't think any other component will use this predicate other than kas.
let's move it to the kas package where its used. 
See this as an example https://github.com/muraee/hypershift/blob/karpenter-userdata/control-plane-operator/controllers/hostedcontrolplane/v2/cno/component.go#L47